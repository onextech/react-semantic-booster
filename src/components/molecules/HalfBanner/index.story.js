import React from 'react';
import { storiesOf } from '@storybook/react';
import { List } from 'semantic-ui-react';
import HalfBanner from '.';
import Block from '../../atoms/Block';
import { testImage2 } from '../../../../test/placeholders';


export const defaultExample = {
  leftCol: {
    background: {
      color: '#1f40e6',
    },
    header: {
      content: 'Center of Focus',
    },
    textAlign: 'center',
    verticalAlign: 'middle',
  },
  rightCol: {
    background: {
      src: testImage2,
    },
    subheader: {
      content: 'Key Trends',
    },
    header: {
      content: 'We reviewed the entire process of data collection, analysis and evaluation to identify key opportunities for automation and efficient data analysis.',
    },
  },
};

const ListExample = () => (
  <List bulleted size="large" relaxed>
    <List.Item>Designed an integration for realtime financial market data population</List.Item>
    <List.Item>
      Built an automatic chart generation and product table tool
      for effortless market performance tracking
    </List.Item>
    <List.Item>Simplified reporting process with built-in language integrations</List.Item>
  </List>
);

const listBannerExample = {
  leftCol: {
    background: {
      color: '#1f40e6',
    },
    header: {
      content: 'From manual data collation to agile financial analysis in a click of a button.',
    },
  },
  rightCol: {
    body: <ListExample />,
    background: {
      color: '#eff3f7',
    },
    textcolor: '#666',
  },
};

const breakExample = {
  leftCol: {
    background: {
      src: testImage2,
      overlay: false,
    },
  },
  rightCol: {
    background: {
      color: '#1f40e6',
    },
    header: {
      content: 'With no digital capabilities and no value proposition for digital talent, the company needed to reinvent itself.',
    },
  },
};

const altTextExample = {
  leftCol: {
    background: {
      src: testImage2,
      overlay: false,
    },
  },
  rightCol: {
    background: {
      color: '#1f40e6',
    },
    header: {
      content: 'Keeping it real',
    },
    body: <p>
      With no digital capabilities and no value proposition for digital talent,
      the company needed to reinvent itself.
      We reviewed the entire process of data collection,
      analysis and evaluation to identify key opportunities for
      automation and efficient data analysis.
    </p>,
  },
};

storiesOf('HalfBanner', module)
  .add('Default', () => (
    <div>
      <Block spacer={0} vertical><HalfBanner {...listBannerExample} /></Block>
      <Block spacer={0} vertical><HalfBanner {...breakExample} /></Block>
      <Block spacer={0} vertical><HalfBanner {...defaultExample} /></Block>
      <Block spacer={0} vertical><HalfBanner {...altTextExample} /></Block>
    </div>
  ));
