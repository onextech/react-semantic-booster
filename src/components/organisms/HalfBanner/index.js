import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { Grid } from 'semantic-ui-react';
import { MediaCss } from '../../../utils/responsive';
import Copy from '../../molecules/Copy/index';
import { verticalAlignFlexCssMap } from '../../../utils/constants';


const StyledGrid = styled(Grid)`
  &.ui.grid.inverted {
    color: white;
  }
`;

const Col = styled(Grid.Column)`
  ${MediaCss.min.sm`height: 100%;`}
  display: flex;
  font-size: 120%;
  
  // Fix for column verticalAlign with the addition of .content element
  ${({ verticalAlign }) => {
    if (verticalAlign) {
      const vaCssValue = verticalAlignFlexCssMap[verticalAlign];
      return `
        &.column.aligned.${verticalAlign} .content {
          height: 100%;
          display: flex;
          align-items: ${vaCssValue};
        }
      `;
    }
    return false;
  }}
  
  // textColor
  ${({ textcolor }) => {
    if (textcolor) {
      return `&.column { color: ${textcolor}; }`;
    }
    return false;
  }}
  
  // background
  ${(props) => {
    if (props.background) {
      const {
        src,
        repeat,
        position,
        opacity,
        fromColor,
        toColor,
        size,
        attachment,
        color,
        overlay = true,
      } = props.background;
      const defaultOpacity = 0.8;
      const linearOpacity = overlay ? (opacity || defaultOpacity) : 0;
      const defaultLinearColor = 'rgba(0,0,0,' + linearOpacity + ')'; // eslint-disable-line
      const linearFrom = fromColor || defaultLinearColor;
      const linearTo = toColor || defaultLinearColor;
      return `
        &.column {
          background-color: ${color || 'black'};
          ${src && `
            background: linear-gradient(${linearFrom}, ${linearTo}), url('${src}');
            background-position: ${position || 'center'};
            background-repeat: ${repeat || 'no-repeat'};
            background-size: ${size || 'cover'};
            background-attachment: ${attachment || 'scroll'};`}
        }`;
    }
    return false;
  }}
  `;

const ColumnContainer = styled.div`
  width: 75%;
  margin: 0 auto;
  padding: 3em 0;
  p {
    line-height: 1.6;
  }
  `;

const HalfBanner = ({
  rightCol,
  leftCol,
  grid,
}) => (
  <StyledGrid stackable inverted {...grid}>
    <Grid.Row columns="equal">
      {
        [leftCol, rightCol].map((col, i) => (
          <Col {...col} key={['left', 'right'][i]}>
            <ColumnContainer className="content">
              <Copy {...col.copy} />
            </ColumnContainer>
          </Col>
        ))
      }
    </Grid.Row>
  </StyledGrid>
);

HalfBanner.propTypes = {
  rightCol: PropTypes.object.isRequired,
  leftCol: PropTypes.object.isRequired,
  grid: PropTypes.object, // sui grid props
};

export default HalfBanner;
