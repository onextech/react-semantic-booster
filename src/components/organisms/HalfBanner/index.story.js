import React from 'react';
import { storiesOf } from '@storybook/react';
import { List } from 'semantic-ui-react';
import HalfBanner from './index';
import Block from '../../atoms/Block/index';
import { testImage2 } from '../../../../test/placeholders';


export const defaultExample = {
  leftCol: {
    background: {
      color: '#1f40e6',
    },
    copy: {
      header: {
        content: 'Center of Focus',
      },
    },
    textAlign: 'center',
    verticalAlign: 'middle',
  },
  rightCol: {
    background: {
      src: testImage2,
    },
    copy: {
      subheader: {
        content: 'Key Trends',
      },
      header: {
        content: 'We reviewed the entire process of data collection, analysis and evaluation to identify key opportunities for automation and efficient data analysis.',
      },
    },
  },
};

const ListExample = () => (
  <List bulleted size="large" relaxed>
    <List.Item>Designed an integration for realtime financial market data population</List.Item>
    <List.Item>
      Built an automatic chart generation and product table tool
      for effortless market performance tracking
    </List.Item>
    <List.Item>Simplified reporting process with built-in language integrations</List.Item>
  </List>
);
export const listBannerExample = {
  leftCol: {
    background: {
      color: '#1f40e6',
    },
    copy: {
      header: {
        content: 'From manual data collation to agile financial analysis in a click of a button.',
      },
    },
  },
  rightCol: {
    copy: {
      content: <ListExample />,
      color: '#666',
    },
    background: {
      color: '#eff3f7',
    },
  },
};

export const breakExample = {
  leftCol: {
    background: {
      src: testImage2,
      overlay: false,
    },
  },
  rightCol: {
    background: {
      color: '#1f40e6',
    },
    copy: {
      header: {
        content: 'With no digital capabilities and no value proposition for digital talent, the company needed to reinvent itself.',
      },
    },
  },
};

export const altTextExample = {
  leftCol: {
    background: {
      src: testImage2,
      overlay: false,
    },
  },
  rightCol: {
    background: {
      color: '#1f40e6',
    },
    copy: {
      header: {
        content: 'Keeping it real',
      },
      body: 'With no digital capabilities and no value proposition for digital talent, the company needed to reinvent itself. We reviewed the entire process of data collection, analysis and evaluation to identify key opportunities for automation and efficient data analysis.',
    },
  },
};

storiesOf('HalfBanner', module)
  .add('Default', () => (
    <div>
      <Block spacer={0} vertical><HalfBanner {...listBannerExample} /></Block>
      <Block spacer={0} vertical><HalfBanner {...breakExample} /></Block>
      <Block spacer={0} vertical><HalfBanner {...defaultExample} /></Block>
      <Block spacer={0} vertical><HalfBanner {...altTextExample} /></Block>
    </div>
  ));
