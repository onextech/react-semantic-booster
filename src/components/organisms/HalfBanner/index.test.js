import React from 'react';
import renderer from 'react-test-renderer';
import HalfBanner from './index';
import { defaultExample } from './index.story';

describe('<HalfBanner>', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<HalfBanner {...defaultExample} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
