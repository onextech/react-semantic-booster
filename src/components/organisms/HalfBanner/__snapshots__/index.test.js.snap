// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HalfBanner> renders correctly 1`] = `
<div
  className="ui inverted stackable grid sc-bdVaJa aLVcX"
>
  <div
    className="equal width row"
  >
    <div
      background={
        Object {
          "color": "#1f40e6",
        }
      }
      className="center aligned middle aligned column sc-bwzfXH cGsBMB"
      header={
        Object {
          "content": "Center of Focus",
        }
      }
    >
      <div
        className="content sc-htpNat leDtIf"
      >
        <h5
          className="ui huge inverted header sc-bxivhb jPPUKp"
        >
          Center of Focus
        </h5>
      </div>
    </div>
    <div
      background={
        Object {
          "src": "https://images.pexels.com/photos/196652/pexels-photo-196652.jpeg?w=940&h=650&dpr=2&auto=compress&cs=tinysrgb",
        }
      }
      className="column sc-bwzfXH knqdmQ"
      header={
        Object {
          "content": "We reviewed the entire process of data collection, analysis and evaluation to identify key opportunities for automation and efficient data analysis.",
        }
      }
      subheader={
        Object {
          "content": "Key Trends",
        }
      }
    >
      <div
        className="content sc-htpNat leDtIf"
      >
        <h6
          className="ui tiny inverted sub header sc-ifAKCX jVuaxL"
        >
          Key Trends
        </h6>
        <h5
          className="ui huge inverted header sc-bxivhb jPPUKp"
        >
          We reviewed the entire process of data collection, analysis and evaluation to identify key opportunities for automation and efficient data analysis.
        </h5>
      </div>
    </div>
  </div>
</div>
`;
